File: backend/components/metiers/threed/tests_centralises/legacy/samples/rgpd/test_rgpd_sample.js
// Test ultra avancé pour rgpd_sample.js
const rgpdSample = require('../../../legacy/samples/rgpd/rgpd_sample');

describe('RGPD Sample', () => {
  it('doit exporter un utilisateur', () => {
    const result = rgpdSample('alice');
    expect(result.user).toBe('alice');
    expect(result.export).toBe(true);
    expect(typeof result.date).toBe('string');
  });
  it('doit exporter un utilisateur vide', () => {
    const result = rgpdSample('');
    expect(result.user).toBe('');
    expect(result.export).toBe(true);
    expect(typeof result.date).toBe('string');
  });
  it('doit exporter un utilisateur null', () => {
    const result = rgpdSample(null);
    expect(result.user).toBeNull();
    expect(result.export).toBe(true);
    expect(typeof result.date).toBe('string');
  });
  it('doit auditer le format de date ISO', () => {
    const result = rgpdSample('bob');
    expect(result.date).toMatch(/\d{4}-\d{2}-\d{2}T/);
  });
  // ...autres tests avancés, edge cases, conformité RGPD...
});

File: backend/components/metiers/threed/tests_centralises/legacy/samples/rgpd/test_rgpd_sample.py
"""
Test ultra avancé pour rgpd_sample.py
"""

from threed.legacy.samples.rgpd.rgpd_sample import rgpd_sample


def test_rgpd_export():
    result = rgpd_sample("alice")
    assert result["user"] == "alice"
    assert result["export"] is True
    assert isinstance(result["date"], str)


def test_rgpd_export_empty_user():
    result = rgpd_sample("")
    assert result["user"] == ""
    assert result["export"] is True
    assert isinstance(result["date"], str)


def test_rgpd_export_none_user():
    result = rgpd_sample(None)
    assert result["user"] is None
    assert result["export"] is True
    assert isinstance(result["date"], str)


def test_rgpd_export_audit_date_format():
    import re

    result = rgpd_sample("bob")
    # Vérifie format ISO 8601
    assert re.match(r"\d{4}-\d{2}-\d{2}T", result["date"])

# ...autres tests avancés, edge cases, conformité RGPD...
